contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,571 @@
+  
+  Specific use cases around improved usage of storage consumption are included in the use cases listed below as follows:
+  * An admin wishes to expose a block volume to be consumed as a block volume for the user  
+  * A user wishes to utilitze block storage to fully realize the performance of an application tuned to using block devices
+  * A user wishes to utilize raw block devices for consumption from a virtual machine
+  * A user wishes to specify an inline volume as a block device in their pod
+  Future use cases include dynamically provisioning and intelligent discovery of existing devices, which this proposal sets the 
+  foundation for more fully developing these methods. 
+  
+  It is importnant to note that when a PV is bound, it is either bound as a raw block device or formatted with a filesystem. Therefore, 
+  the PVC drives the request and intended usage of the device by specifying the volumeMode as part of the API. This design lends itself
+  to future support of dynamic provisioning by also letting the request intiate from the PVC defining the role for the PV. It also allows
+  flexibility in the implementation and storage plugins to determine their support of this feature.
+ 
+# Design Overview
+
+  The proposed design is based on the idea of leveraging well defined concepts for storage in Kubernetes. The consumption and 
+  definitions for the block devices will be driven through the PVC and PV definitions. Along with Storage
+  Resource definitions, this will provide the admin with a consistent way of managing all storage. 
+  The API changes proposed in the following section are minimal with the idea of defining a volumeMode to indicate both the definition
+  and consumption of the devices. Since it's possible to create a volume as a block device and then later consume it by provisioning
+  a filesystem on top, the design requires explicit intent for how the volume will be used.
+  The additional benefit of explicitly defining how the volume is to be consumed will provide a means for indicating the method
+  by which the device should be scrubbed when the claim is deleted, as this method will differ from a raw block device compared to a 
+  filesystem. The ownership and responsibility of defining the rention policy shall be up to the plugin method being utilized and is not 
+  covered in this proposal.
+  
+  The last design point is block devices should be able to be fully restricted by the admin in accordance with how inline volumes 
+  are today. Ideally, the admin would want to be able to restrict either raw-local devices and or raw-network attached devices.
+  
+  To ensure backwards compatibility and a phased transition of this feature, the consensus from the community is to intentionally disable
+  the volumeMode: Block for external provisioners until a suitable implementation for provisioner versioning has been accepted and 
```

Comment:
What about internal provisioners too?

Reply:
@msau42 I don't understand your comment....We are not disabling it for internal provisioners as we are updating them as part of the PR

Reply:
You're updating them not to provision for block devices, or updating them to provision block devices?

Reply:
@childsb updating them to understand volumeMode: Block so they don't default to filesystem

Reply:
per my suggestion further down the thread, i'd leave the in-tree provisioners alone and if we ever want block provisioners, we create new provisioners.

Reply:
@childsb yes they would have to be new provisioners